x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -359,8 +360,13 @@ public void onFailure(Exception e) {
         }
     }
 
-    private void failIfSkipUnavailableFalse(Exception e, String cluster, ActionListener<LookupResponse> lookupListener) {
-        if (ExceptionsHelper.isRemoteUnavailableException(e) && remoteClusterService.isSkipUnavailable(cluster).orElse(true)) {
+    private void failIfSkipUnavailableFalse(
+        Exception e,
+        String cluster,
+        EsqlExecutionInfo executionInfo,
+        ActionListener<LookupResponse> lookupListener
+    ) {
```

Comment:
Here again (as below) we don't really need neither `cluster` nor `EsqlExecutionInfo` but only boolean `executionInfo.shouldSkipOnFailure(cluster)`. Maybe we could simplify it by calculating this boolean above and pass it to the methods that need it? Just a suggestion.